zrml/swaps/src/lib.rs

```diff
@@ -398,6 +429,9 @@ mod pallet {
                     )?.saturated_into();
                     ensure!(asset_amount_out >= min_asset_amount_out, Error::<T>::LimitOut);
 
+                    T::LiquidityMining::remove_shares(&who, &pool.market_id, asset_amount_out);
+                    T::LiquidityMining::add_shares(who_clone, pool.market_id, asset_amount_in);
```

Comment:
I am a bit confused here. So everyone who trades Asset A OUT for Asset B IN, automatically becomes a liquidity provider and earns amount IN shares? `add_shares` and `remove_shares` in liquidity provider pallet also need some documentation. What are those shares? LP shares? Assets?

Reply:
My bad. These functions receive pool shares instead of asset amounts.
Another reason to create smaller PRs: They are also easier to review and less error-prone for the author

Reply:
Wait a second. `asset_amount_in` should be `shares_amount_in`...

Reply:
No, only adding and removing liquidity is considered for liquidity mining.

Reply:
Thanks guys. Just removed liquidity functions from swapping operations